cpp/daal/include/sycl/internal/math/mkl_lapack.h

```diff
@@ -67,6 +73,20 @@ struct MKLPotrf
         return status;
     }
 
+    services::Status operator()(const math::UpLo uplo, const size_t n, services::Buffer<algorithmFPType> & a, const size_t lda,
+                                const std::int64_t scratchpad_size)
+    {
+        cl::sycl::buffer<algorithmFPType, 1> scratchpad_buffer { cl::sycl::range<1>(scratchpad_size) };
+        return this->operator()(uplo, n, a, lda, scratchpad_buffer);
+    }
+
+    services::Status operator()(const math::UpLo uplo, const size_t n, services::Buffer<algorithmFPType> & a, const size_t lda)
+    {
+        const fpk::uplo uplomkl                    = uplo == math::UpLo::Upper ? fpk::uplo::upper : fpk::uplo::lower;
+        const std::int64_t minimal_scratchpad_size = fpk::lapack::potrf_scratchpad_size<algorithmFPType>(_queue, uplomkl, n, lda);
+        return this->operator()(uplo, n, a, lda, minimal_scratchpad_size);
```

Comment:
Is `potrf_scratchpad_size` expensive operation? The problem is one of the operator overloadings compares buffers size with the value returned by `potrf_scratchpad_size`. In that function, we use `potrf_scratchpad_size` to determine required buffer size, so check for error in the overloading just adds overhead.

If `potrf_scratchpad_size` is cheap, ignore this comment.

Reply:
It seems to be very cheap.

Reply:
I've checked an assembly at it's pretty compact and computationally cheap function